### PR TITLE
beego v2 session refactor

### PR DESCRIPTION
**Pull request by MinerYang**
Signed-off-by: yminer <yminer@vmware.com>

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [ ] Well Written Title and Summary of the PR
- [ ] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [ ] Accepted the DCO. Commits without the DCO will delay acceptance.
- [ ] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: src/core/session/session.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ func (rp *Provider) SessionInit(maxlifetime int64, url string) (err error) {
 }
 
 // SessionRead read redis session by sid
-func (rp *Provider) SessionRead(sid string) (session.Store, error) {
+func (rp *Provider) SessionRead(ctx context.Context, sid string) (session.Store, error) {
 	kv := make(map[interface{}]interface{})
 	err := rp.c.Fetch(context.TODO(), sid, &kv)
```
Comment (chlins) [inline src/core/session/session.go:122]: Pls replace the `context.TODO` to the function argument `ctx`.

File: src/core/session/session.go [hunk 2]
```diff
@@ -129,14 +129,13 @@ func (rp *Provider) SessionRead(sid string) (session.Store, error) {
 }
 
 // SessionExist check redis session exist by sid
-func (rp *Provider) SessionExist(sid string) bool {
-	return rp.c.Contains(context.TODO(), sid)
+func (rp *Provider) SessionExist(ctx context.Context, sid string) (bool, error) {
+	return rp.c.Contains(context.TODO(), sid), nil
```
Comment (chlins) [inline src/core/session/session.go:133]: ditto

File: src/core/session/session.go [hunk 3]
```diff
@@ -129,14 +129,13 @@ func (rp *Provider) SessionRead(sid string) (session.Store, error) {
...
 }
 
 // SessionRegenerate generate new sid for redis session
-func (rp *Provider) SessionRegenerate(oldsid, sid string) (session.Store, error) {
-	ctx := context.TODO()
-	if !rp.SessionExist(oldsid) {
+func (rp *Provider) SessionRegenerate(ctx context.Context, oldsid, sid string) (session.Store, error) {
+	if isExist, _ := rp.SessionExist(context.Background(), oldsid); !isExist {
```
Comment (chlins) [inline src/core/session/session.go:138]: ditto

**Review by chlins — APPROVED** [review]
lgtm

File: src/core/session/session.go [hunk 4]
```diff
@@ -164,20 +163,20 @@ func (rp *Provider) SessionRegenerate(oldsid, sid string) (session.Store, error)
 		}
 	}
 
-	return rp.SessionRead(sid)
+	return rp.SessionRead(context.Background(), sid)
```
Comment (chlins) [inline src/core/session/session.go:166]: pls use ctx from method parameter

**Review by wy65701436 — APPROVED** [review]
lgtm

**Pull request merged**